cosmoz-data-nav.js

```diff
@@ -477,16 +477,44 @@
 					index = items.findIndex(item => this._getItemId(item) === prevId);
 				}
 
-				// if still not found, remain on the selected index, but force re-render
+				// if still not found, remain on the selected index
 				if (index === -1) {
-					return this._updateSelected();
+					index = this.selected;
 				}
+				this._realignElements(index);
 			}
-
 			// update selected or force re-render if selected did not change
 			return this.selected === index ? this._updateSelected() : this.selected = index;
 		}
 
+		_realignElements(index) {
+			const elements = this._elements,
+				element = this._getElement(index),
+				item = this.items[index];
+			if (this.isIncompleteFn(item) || element.item === item) {
+				return;
+			}
+			const renderedElement = this._elements.find(el => !this.isIncompleteFn(el.item) && this._getItemId(el.item) === this._getItemId(item));
+			if (!renderedElement) {
+				return;
+			}
+			const elementIndex = elements.indexOf(element);
+			const renderedIndex = elements.indexOf(renderedElement);
+			if (elementIndex === renderedIndex) {
```

Comment:
I think we can (should?) drop this check unless we can also add a comment why `element.item !== item && elementIndex === renderedIndex` can occur.